vcd_cli/user.py

```diff
@@ -143,3 +148,34 @@ def create(ctx, user_name, password, role_name, full_name, description, email,
         stdout('User \'%s\' is successfully created.' % u.get('name'), ctx)
     except Exception as e:
         stderr(e, ctx)
+
+@user.command(short_help='delete an user in current organization')
+@click.pass_context
+@click.argument('user_name',
+                metavar='<user_name>',
+                required=True)
+@click.option('-y',
+              '--yes',
+              is_flag=True,
+              callback=abort_if_false,
+              expose_value=False,
+              prompt='Are you sure you want to delete the user?')
+def delete(ctx, user_name):
+    """
+    Deletes an user from the current organization. Also deletes all assests (e.g. vApps, vms, catalogs etc.) owned by the user. However if any of those vApps are running, the command will result in error.
+
+\b
+    Examples
+        vcd user delete 'my user'
+           deletes user with username "my user" from the current organization.
+
+    """  # NOQA
+    try:
+        client = ctx.obj['client']
+        org_name = ctx.obj['profiles'].get('org')
+        in_use_org_href = ctx.obj['profiles'].get('org_href')
+        org = Org(client, in_use_org_href, org_name == 'System')
```

Comment:
Consider using is_sysAdmin() utils method, which is more descriptive to the code readers.